x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -56,28 +69,75 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     skip: !expandedEvent.eventId,
   });
 
+  const [hostIsolationPanelOpen, setHostIsolationPanel] = useState(false);
+
+  const showAlertDetails = useCallback(() => {
+    setHostIsolationPanel(false);
+  }, []);
+
+  const showHostIsolationPanel = useCallback(() => {
+    setHostIsolationPanel(true);
+  }, []);
+
   const isAlert = some({ category: 'signal', field: 'signal.rule.id' }, detailsData);
 
+  const findEndpointAlert = find({ category: 'agent', field: 'agent.type' }, detailsData)?.values;
```

Comment:
As detailsData is a dynamic, could you put this inside a useMemo()?

Reply:
Just in case to avoid performance issues if detailsData.values is too large and it has to be calculated on each render